### PR TITLE
dxguids: guard Direct3D with __d3d12_h__

### PR DESCRIPTION
**Pull request by lygstate**
By doing this so that WINADAPTER_IID can be used without include of `<directx/d3d12.h>`

Signed-off-by: Yonggang Luo <luoyonggang@gmail.com>

### CODE REVIEW

**Comment by walbourn** [comment]
There are merged in https://github.com/microsoft/DirectX-Headers/commit/4c2339cb02ae6d7dc7f6e56437ccff038ec8f126

**Comment by lygstate** [comment]
@walbourn This is a different PR.

**Comment by lygstate** [comment]
Pelease reopen it

File: include/dxguids/dxguids.h [hunk 1]
```diff
@@ -55,6 +55,7 @@ template <> constexpr GUID uuidof<InterfaceName>() \
 #endif /* _WIN32 */
 
 // Direct3D
+#ifdef __d3d12_h__
```
Comment (jenatali) [inline include/dxguids/dxguids.h:58]: Yeah, I see where you're going with this. The original intention here was to always include this after D3D12, but I'm remembering that you're trying to get the DXIL header building, which relies on the same IID infrastructure/defines. Given that, yeah I think this is okay - I was tempted to suggest adding a `#warning` if none of these headers were included, but I think you do have a valid scenario here.

**Review by jenatali — APPROVED** [review]

File: include/dxguids/dxguids.h (hunk 1 above)
Comment (lygstate) [reply include/dxguids/dxguids.h:58]: I think the `#warning` should be in `d3d12.h` and alike files.
That's makes more sense.

Comment (lygstate) [reply include/dxguids/dxguids.h:58]: Before this patch, the compiling error is rahter annoy, too long to debugging.

**Pull request merged**